build-aux/snap/local/rename-dynamic-loader-variables.patch

```diff
@@ -0,0 +1,40 @@
+diff -ur glibc-2.35.old/elf/dl-environ.c glibc-2.35/elf/dl-environ.c
+--- glibc-2.35.old/elf/dl-environ.c	2024-09-16 12:34:26.933137126 +0200
++++ glibc-2.35/elf/dl-environ.c	2024-09-16 12:45:01.558071825 +0200
+@@ -31,10 +31,21 @@
+ 
+   while (*current != NULL)
+     {
+-      if (__builtin_expect ((*current)[0] == 'L', 0)
+-	  && (*current)[1] == 'D' && (*current)[2] == '_')
++      if (__builtin_expect ((*current)[0] == 'S', 0)
```

Comment:
would environment variables like LD_DEBUG now need to be prefixed with SNAPD_ too?

Reply:
Yes.

Reply:
Looked at process_envvars() in https://git.launchpad.net/ubuntu/+source/glibc/tree/elf/rtld.c?h=ubuntu/jammy-updates#n2673, further processing only relies on the variable name without the LD_ (or in this case SNAPD_LD_) prefix, so this looks good.

ld.so strips unsecure environment variables like LD_PRELOAD and the like, does it make sense to extend the list in https://git.launchpad.net/ubuntu/+source/glibc/tree/sysdeps/generic/unsecvars.h?h=ubuntu/jammy-updates to include SNAPD_ variants as well?

Reply:
Should we also update the strip out list in osutil/env.go?https://github.com/canonical/snapd/blob/master/osutil/env.go#L234

Reply:
I have added the names, instead of just renaming. In case there are some other corner we did not see that it reads the old variables.